award_emojis.go

```diff
@@ -0,0 +1,219 @@
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/%s/%d/award_emoji/%d", url.QueryEscape(project), resource, resourceID, awardID)
+
+	req, err := a.client.NewRequest("POST", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	awardEmoji := new(AwardEmoji)
+	resp, err := a.client.Do(req, &awardEmoji)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return awardEmoji, resp, err
+}
+
+// DeleteAwardEmoji Delete an award emoji on the specified resource
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#delete-an-award-emoji
+func (a *AwardEmojiService) DeleteAwardEmoji(pid interface{}, resource string, resourceID, awardID int, options ...OptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/%s/%d/award_emoji/%d", url.QueryEscape(project), resource, resourceID, awardID)
+
+	req, err := a.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+	return a.client.Do(req, nil)
+}
+
+// ListAwardEmojiOnNote list single note's award emoji
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-a-note-39-s-award-emoji
+func (a *AwardEmojiService) ListAwardEmojiOnNote(pid interface{}, issueIID, noteID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/issues/%d/notes/%d/award_emoji", url.QueryEscape(project), issueIID, noteID)
+
+	req, err := a.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var awardEmoji []*AwardEmoji
+	resp, err := a.client.Do(req, &awardEmoji)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return awardEmoji, resp, err
+}
+
+// GetSingleNoteAwardEmoji get single note's award emoji
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#get-single-note-s-award-emoji
+func (a *AwardEmojiService) GetSingleNoteAwardEmoji(pid interface{}, issueIID, noteID, awardID int, options ...OptionFunc) (*AwardEmoji, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/issues/%d/notes/%d/award_emoji/%d", url.QueryEscape(project), issueIID, noteID, awardID)
+
+	req, err := a.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	awardEmoji := new(AwardEmoji)
+	resp, err := a.client.Do(req, &awardEmoji)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return awardEmoji, resp, err
+}
+
+// CreateAwardEmojiOnNote award emoji on a note
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#award-a-new-emoji-on-a-note
+func (a *AwardEmojiService) CreateAwardEmojiOnNote(pid interface{}, issueIID, noteID int, options ...OptionFunc) (*AwardEmoji, *Response, error) {
```

Comment:
Probably want to add a `DeleteAwardEmojiOnNote` to complete the set.

Reply:
Added